src/Makefile.am

```diff
@@ -778,8 +782,42 @@ bitcoin_util_LDADD = \
 #
 
 # bitcoin-chainstate binary #
-bitcoin_chainstate_SOURCES = \
-  bitcoin-chainstate.cpp \
+bitcoin_chainstate_SOURCES = bitcoin-chainstate.cpp
+bitcoin_chainstate_CPPFLAGS = $(AM_CPPFLAGS) $(BITCOIN_INCLUDES)
+bitcoin_chainstate_CXXFLAGS = $(AM_CXXFLAGS) $(PIE_FLAGS)
+bitcoin_chainstate_LDFLAGS = $(AM_LDFLAGS) $(LIBTOOL_APP_LDFLAGS) $(PTHREAD_FLAGS)
+bitcoin_chainstate_LDADD = $(LIBBITCOINKERNEL)
+#
+
+# bitcoinconsensus library #
+if BUILD_BITCOIN_LIBS
+lib_LTLIBRARIES += $(LIBBITCOINCONSENSUS)
+
+include_HEADERS = script/bitcoinconsensus.h
+libbitcoinconsensus_la_SOURCES = support/cleanse.cpp $(crypto_libbitcoin_crypto_base_la_SOURCES) $(libbitcoin_consensus_a_SOURCES)
+
+libbitcoinconsensus_la_LDFLAGS = $(AM_LDFLAGS) -no-undefined
+libbitcoinconsensus_la_LIBADD = $(LIBSECP256K1)
+libbitcoinconsensus_la_CPPFLAGS = $(AM_CPPFLAGS) -I$(builddir)/obj -I$(srcdir)/secp256k1/include -DBUILD_BITCOIN_INTERNAL
+libbitcoinconsensus_la_CXXFLAGS = $(AM_CXXFLAGS) $(PIE_FLAGS)
+
+endif
+#
+
+# bitcoinkernel library #
+if BUILD_BITCOIN_KERNEL_LIB
+lib_LTLIBRARIES += $(LIBBITCOINKERNEL)
+
+libbitcoinkernel_la_LDFLAGS = $(AM_LDFLAGS) $(LIBTOOL_APP_LDFLAGS) $(PTHREAD_FLAGS)
```

Comment:
I've been going down another deep rabbit hole with this, and it is surprisingly hard to create a shared mingw-w64 dll with `libstdc++`, `libpthread`, etc statically embedded...

What I'm going to do is just leave it as a static library for now, so that we can make progress on the main thrust of the project, and work on getting the build correct in the mean time. It is very likely that  `LIBTOOL_IS_A_FOOL`...

Reply:
If this is just creating another static library now, I don't really understand why it's still so complicated? Can't you just add the static library, and not do all the libtool stuff?

Reply:
> If this is just creating another static library now, I don't really understand why it's still so complicated? Can't you just add the static library, and not do all the libtool stuff?

A new comment https://github.com/dongcarl/bitcoin/blob/3961a789735c874ed9b3ff4bbd64c2b850282c07/src/Makefile.am#L811-L813 mentions plans to build libbitcoinkernel as a shared library instead of a static library in the future. IIUC, libtool is needed to do that to make sure static libraries linked into the dynamic library are built with correct PIC flags.

Reply:
Right, but why not leave the libtool / .la commits until the future PR that does build libbitcoinkernel as a shared library?

Reply:
> Right, but why not leave the libtool / .la commits until the future PR that does build libbitcoinkernel as a shared library?

Dude, I don't know. I think libbitcoinkernel is just big project and we're trying to make incremental progress.

These commits are simple, basically just replacing _a with _la, and would be annoying to maintain separately because they'll conflict with pretty much every other change to these files. They also clean up some obsolete things, make the libraries compatible with dynamic linking generically (not specific to libbitcoinkernel), and document the libraries varying abilities to be dynamically linked themselves. So they are good changes.

It'd be fine to move these commits into separate PRs if that would _actually_ help get this PR reviewed. But otherwise I think questions about how PRs are subdivided are not that useful, and that it would be a loss and obstacle to progress if these changes had delayed indefinitely until they were part of some more comprehensive "future PR".

Reply:
With the latest push, we build shared libraries for all platforms except Windows! 🎉 

> I think libbitcoinkernel is just big project and we're trying to make incremental progress.
☺️